xarray/util/generate_reductions.py

```diff
@@ -45,9 +46,9 @@ class {obj}{cls}Reductions:
     def reduce(
         self,
         func: Callable[..., Any],
-        dim: None | Hashable | Sequence[Hashable] = None,
+        dim: str | Iterable[Hashable] | None = None,
```

Comment:
```suggestion
        dim: T_Dim = None,
```
Should this become a variable? It should make it easier adding the typing in `flox` easier as well.